src/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1427,11 +1427,13 @@ public static String validateCanal(final Route route, final Collection<Unit> uni
     final Set<CanalAttachment> canalAttachments = CanalAttachment.get(territory);
     for (final CanalAttachment canalAttachment : canalAttachments) {
       if (!isCanalOnRoute(canalAttachment, route, data)) {
-        continue;
+        continue; // Only check canals that are on the route
       }
       failureMessage = canPassThroughCanal(canalAttachment, units, player, data);
```

Comment:
It's not that I personally have trouble reading failureMessage... The idea is that it should not require too much effort at all to understand a section of code.
But. we have a SRP violation with the variable name `failureMessage`. The variable 'failureMessage' is *not* always a failure message.